src/tactic/suggest.lean

```diff
@@ -109,40 +111,40 @@ do env ← get_env,
 
 /--
 Apply the lemma `e`, then attempt to close all goals using
-`solve_by_elim { discharger := discharger }`, failing if `close_goals = tt`
+`solve_by_elim { ..opt }`, failing if `close_goals = tt`
 and there are any goals remaining.
 -/
 -- Implementation note: as this is used by both `library_search` and `suggest`,
 -- we first run `solve_by_elim` separately on a subset of the goals,
 -- whether or not `close_goals` is set,
 -- and then if `close_goals = tt`, require that `solve_by_elim { all_goals := tt }` succeeds
 -- on the remaining goals.
-meta def apply_and_solve (close_goals : bool) (discharger : tactic unit) (e : expr) : tactic unit :=
+meta def apply_and_solve (close_goals : bool) (opt : opt := { }) (e : expr) : tactic unit :=
 apply e >>
 -- Phase 1
 -- Run `solve_by_elim` on each "safe" goal separately, not worrying about failures.
 -- (We only attempt the "safe" goals in this way in Phase 1. In Phase 2 we will do
 -- backtracking search across all goals, allowing us to guess solutions that involve data, or
 -- unify metavariables, but only as long as we can finish all goals.)
-try (any_goals (independent_goal >> solve_by_elim { discharger := discharger })) >>
+try (any_goals (independent_goal >> solve_by_elim { ..opt })) >>
```

Comment:
Just write `opt`, not `{ ..opt }`. Why did you want to do that?

Reply:
No good reason, `opt` is better.